### ISSUE TITLE
Unable to compile on Mac M1 architecture

### ISSUE THREAD

**Issue by MatthewDaggitt** [haskell-unordered-containers/hashable#261]
We're struggling to compile `hashable` on a Mac M1 laptop:
```
macOS Monterey
Version 12.1(21C52)

MacBook Pro (13-inch, M1, 2020)
Chip Apple M1
Memory 16 GB
``` 
The error message we get is:
```
Failed to build hashable-1.4.1.0.
Build log (
/Users/marcocasadio/.cabal/logs/ghc-9.0.2/hshbl-1.4.1.0-865f38d3.log ):
Configuring library for hashable-1.4.1.0..
Preprocessing library for hashable-1.4.1.0..
Building library for hashable-1.4.1.0..
[1 of 7] Compiling Data.Hashable.Imports ( src/Data/Hashable/Imports.hs, dist/build/Data/Hashable/Imports.o, dist/build/Data/Hashable/Imports.dyn_o )
[2 of 7] Compiling Data.Hashable.LowLevel ( src/Data/Hashable/LowLevel.hs, dist/build/Data/Hashable/LowLevel.o, dist/build/Data/Hashable/LowLevel.dyn_o )

In file included from /var/folders/9_/9l6qz02n6yz4qlgqvdfr1f500000gn/T/ghc9908_0/ghc_20.c:4:0: error:
    

In file included from /Users/marcocasadio/.ghcup/ghc/9.0.2/lib/ghc-9.0.2/lib/../lib/aarch64-osx-ghc-9.0.2/rts-1.0.2/include/ffi.h:66:0: error:
    

/Users/marcocasadio/.ghcup/ghc/9.0.2/lib/ghc-9.0.2/lib/../lib/aarch64-osx-ghc-9.0.2/rts-1.0.2/include/ffitarget.h:6:10: error:
     fatal error: 'ffitarget_arm64.h' file not found
  |
6 | #include "ffitarget_arm64.h"
  |          ^
#include "ffitarget_arm64.h"
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
`gcc' failed in phase `C Compiler'. (Exit code: 1)
cabal: Failed to build hashable-1.4.1.0 (which is required by
exe:vehicle-new-golden-test from vehicle-0.1.0 and exe:vehicle from
vehicle-0.1.0). See the build log above for details.
```

Do you have any advice?

**Comment by phadej** [comment]
This looks like GHC issue (that `#include` is generated by GHC). Open an issue on ](https://gitlab.haskell.org/ghc/ghc/

**Comment by MatthewDaggitt** [comment]
Thanks @phadej! 

For those with the same problem, the relevant GHC issue is [here](https://gitlab.haskell.org/ghc/ghc/-/issues/20592) and the problem was apparently fixed in GHC 9.2.3.
